File: src/main/scala/fpgrowth/Itemset.scala
package fpgrowth

import scala.collection.immutable
import scala.collection.mutable.{ArrayBuffer, ListBuffer}

class Itemset[T](var items: ArrayBuffer[T], var support: Long) extends Ordered[Itemset[T]] {

  def this() = {
    this(null, 0)
    this.items = ArrayBuffer.empty[T]
    this.support = 0
  }

  /**
    * @param items the items to set
    */
  def setItems(items: ArrayBuffer[T]) {
    this.items = items
  }

  @throws(classOf[CloneNotSupportedException])
  override def clone: Itemset[T] = {
    super.clone.asInstanceOf[Itemset[T]]
  }

  def addItem(item: T) {
    items += item
  }

  override def toString: String = {
    val string: StringBuilder = new StringBuilder
    string.append("(")
    items.foreach { string.append(_)}
    string.append("):" + support)
    string.toString
  }

  override def compare(o: Itemset[T]): Int = {
    this.support compare o.support
  }
}

File: src/main/scala/fpgrowth/Itemset.java
//package fpgrowth
package fpgrowth;

import java.util.ArrayList;
import java.util.List;


public class Itemset implements Comparable<Itemset> {
	private List<Item> items;
	private int support;
	
	public Itemset() {
		items = new ArrayList<Item>();
		support = 0;
	}
	/**
	 * @return the items
	 */
	public List<Item> getItems() {
		return items;
	}
	/**
	 * @param items the items to set
	 */
	public void setItems(List<Item> items) {
		this.items = items;
	}
	/**
	 * @return the support
	 */
	public int getSupport() {
		return support;
	}
	/**
	 * @param support the support to set
	 */
	public void setSupport(int support) {
		this.support = support;
	}

	public Itemset clone() throws CloneNotSupportedException {
		return (Itemset)super.clone();
	}
	
	public void addItem(Item item) {
		items.add(item);
	}
	
	public String toString() {
		
		StringBuilder string = new StringBuilder();
		string.append("(");
		for(int i = 0; i < items.size(); i++) {
			string.append(items.get(i).toString());
		}
		string.append("):" + support);
		
		return string.toString();
	}

    public int compareTo(Itemset o) {
        return o.support - this.support;
    }
}
